packages/rsocket-tcp-client/src/TcpClientTransport.ts

```diff
@@ -0,0 +1,41 @@
+import net from "net";
+import { ClientTransport, DuplexConnection } from "@rsocket/rsocket-types";
+import { TcpDuplexConnection } from "./TcpDuplexConnection";
+
+export type ClientOptions = {
+  connectionOptions: net.NetConnectOpts;
+};
```

Comment:
`connectionOptions` could maybe just be spread on the top-level options object, rather than nested, however, I thought it might be nice to segment the connection level options from other options that may be related to debug logging, etc. Ultimately I think we'll just want the various transports to have a consistent API.